src/main/java/com/example/demo/PatientController.java

```diff
@@ -0,0 +1,67 @@
+package com.example.demo;
+
+import lombok.AllArgsConstructor;
+import org.springframework.web.bind.annotation.*;
+
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Optional;
+
+@RestController
+@RequestMapping("/patients")
+@AllArgsConstructor
+public class PatientController {
+    private final List<Patient> patients;
+
+    @GetMapping
+    public List<Patient> getAllPatients() {
+        return patients;
+    }
+
+    @GetMapping("/{email}")
+    public Patient getPatientByEmail(@PathVariable String email) {
+        Optional<Patient> findPatient = patients.stream()
+                .filter(patient -> patient.getEmail() != null && patient.getEmail().equals(email))
+                .findFirst();
+
+        if (findPatient.isPresent()) {
```

Comment:
if (findPatient.isEmpty()) {
            throw new RuntimeException("Patient with applied email is not exciting");
         }
      return findPatient.get()

Reply:
Optional<Patient> findPatient = patients.stream()
                .filter(patient -> patient.getEmail() != null && patient.getEmail().equals(email))
                .findFirst();

-> wynies do oddzielnej metody prywatnej i nazwij ja getPatientByEmail(String email)